Libraries/Arduino_BNO055/BNO055_IMU.cpp

```diff
@@ -0,0 +1,138 @@
+#include <BNO055_IMU.h>
+
+void read_data(AxisData *data, DataMode mode)
+{
+    switch (mode) {
+        case GYRO:
+            {
+                uint8_t raw_data[6];
+                i2c_read(I2C_ADDRESS, GYR_DATA_X_LSB, 6, raw_data);
+                data->gyroAxes[0] = ((int16_t)raw_data[1] << 8) | raw_data[0];
```

Comment:
Decision is yours, but my suggestion: might want to create a helper function since this packing behavior shows up a lot. Example:
void pack_raw_data(int16_t* target, uint8_t* raw, int len) { 
    for (int i = 0; i < len; ++i)
        target[i] = *raw++ | (*raw++ << 8); 
}

Reply:
I kind of like how it's spread out, so I'll just make a macro to keep it clean.

Reply:
```cpp
// Macro for unpacking the raw data
#define UNPACK(x,y)    ((((itn16_t)(y)) << 8) | (x))

/* ... */

data->gyroAxes[0] = UNPACK(raw_data[0], raw_data[1]);
```